vent/controller/control_module.py

```diff
@@ -698,45 +637,104 @@ def stop(self):
         if self._save_logs:               # If we kept records, flush the data
             self.dl.close_logfile()
 
+    def interrupt(self):
+        """
+        If a controller seems stuck, this makes a new thread, and starts the main loop.
+        No parameters should have changed.
+        """
+        self._running = threading.Event()           # New thread
+        self._running.clear()
+        self._lock = threading.Lock()
+        self._running.set()
+        self.__thread = threading.Thread(target=self._start_mainloop, daemon=True)
+        try:
+            self.__thread.start()
+        except:
+            pass
+            #TODO RAISE ALERT FOR UI
 
     def is_running(self):
+        self._time_last_contact = time.time()
         # TODO: this should be better thread-safe variable
         return self._running.is_set()
 
+    def get_heartbeat(self):
+        """
+        Returns a heart-beat of the controller, i.e. the internal loop counter
+        """
+        self._time_last_contact = time.time()
+        return self._loop_counter
+
```

Comment:
@sneakers-the-rat here is your heartbeat

Reply:
ty <3